File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.a = deque()
        self.b = deque()

    def push(self, x: int) -> None:
        self.a.append(x)

    def pop(self) -> int:
        while len(self.a) > 1:
            self.b.append(self.a.popleft())
        temp = self.a.popleft()
        while len(self.b):
            self.a.append(self.b.popleft())
        return temp

    def top(self) -> int:
        return self.a[-1]      

    def empty(self) -> bool:
        return len(self.a) == 0
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.kt
class MyStack() {
    private val a = ArrayDeque<Int>()
    private val b = ArrayDeque<Int>()

    fun push(x: Int) {
        a.add(x)
    }

    fun pop(): Int {
        while (a.size > 1) {
            b.add(a.remove())
        }
        val temp = a.remove()
        while (b.size > 0) {
            a.add(b.remove())
        }
        return temp
    }

    fun top(): Int {
        return a.last()
    }

    fun empty(): Boolean {
        return a.size === 0
    }

}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */